examples/scatterplot-renderer.js

```diff
@@ -0,0 +1,148 @@
+import * as d3 from 'd3';
+import { createSvgRenderer } from '../src/renderer';
+
+const DEFAULT_PADDING = 20;
+const DEFAULT_SIZE = 100;
+const DEFAULT_DOTSIZE_RANGE = [1.5, 4];
+const DEFAULT_COLOR_RANGE = d3.schemeDark2;
+const DEFAULT_LINE_COLOR = '#333';
+const DEFAULT_TEXT_COLOR = '#aaa';
+
+const createScatterplotRenderer = ({
+  width = 600,
+  height = 600,
+  x = 'fertilityRate',
+  y = 'lifeExpectancy',
+  size: dotSize = 'population',
+  color: colorProperty = 'region'
+} = {}) => {
+  const svgRenderer = createSvgRenderer({ width, height });
+
+  const createXPos = domain =>
+    d3
+      .scaleLinear()
+      .domain(domain)
+      .rangeRound([DEFAULT_PADDING / 2, DEFAULT_SIZE - DEFAULT_PADDING / 2]);
+
+  const createYPos = domain =>
+    d3
+      .scaleLinear()
+      .domain(domain)
+      .rangeRound([DEFAULT_SIZE - DEFAULT_PADDING / 2, DEFAULT_PADDING / 2]);
+
+  const createZPos = domain =>
+    d3
+      .scaleLinear()
+      .domain(domain)
+      .range(DEFAULT_DOTSIZE_RANGE);
+
+  const createColorScale = domain =>
+    d3
+      .scaleOrdinal()
+      .domain(domain)
+      .range(DEFAULT_COLOR_RANGE);
+
+  const createScatterplot = (xDomain, yDomain, sizeDomain, color) => {
+    const svg = d3
+      .create('svg')
+      .attr('viewBox', `0 0 ${DEFAULT_SIZE} ${DEFAULT_SIZE}`);
+
+    const xPos = createXPos([Math.min(...xDomain), Math.max(...xDomain)]);
```

Comment:
The xPos, yPos, sizeScale, xAxis, yAxis, and colorMap, should be shared between the small multiples. This has two advantages:
1. It's will improve the rendering performance
2. It will ensure that the scatterplots share the same axis. Right now comparison between scatterplots is hard as they are scaled to their individual data.